chainer/training/triggers/early_stopping_trigger.py

```diff
@@ -45,6 +45,7 @@ def __init__(self, check_trigger=(1, 'epoch'), monitor='main/loss',
         self.monitor = monitor
         self.verbose = verbose
         self.already_warning = False
+        self.max_trigger = max_trigger  # TODO(This is confusing)
```

Comment:
How about making `max_trigger` argument accepting any trigger that supports `get_training_length`? Then, we do not need to add `self.max_trigger` and then can make `get_training_length` just delegates to `self._max_trigger.get_training_length()`.

Reply:
Thanks for the great suggestion! I remove `self.max_trigger` and use `self._max_trigger.get_training_length`.